src/ucm/bistro/bistro_rv64.c

```diff
@@ -0,0 +1,105 @@
+/**
+ * Copyright (C) Tactical Computing Labs, LLC. 2022. ALL RIGHTS RESERVED.
+ *
+ * See file LICENSE for terms.
+ */
+
+#ifdef HAVE_CONFIG_H
+#  include "config.h"
+#endif
+
+/* *******************************************************
+ * RISC-V processors family                              *
+ * ***************************************************** */
+#if defined(__riscv)
+
+#include <sys/mman.h>
+#include <dlfcn.h>
+#include <string.h>
+#include <stdlib.h>
+#include <stdint.h>
+#include <stdbool.h>
+#include <assert.h>
+
+#include <ucm/bistro/bistro.h>
+#include <ucm/bistro/bistro_int.h>
+#include <ucm/util/sys.h>
+#include <ucs/sys/math.h>
+#include <ucs/arch/cpu.h>
+#include <ucs/debug/assert.h>
+
+/* Registers numbers to use with the move immediate to register.
+  * The destination register is X31 (highest temporary).
+  * Register X28-X30 are used for block shifting and masking.
+  * Register X0 is always zero
+  */
+ #define X31 31
+ #define X30 30
+ #define X0  0
+
+ /**
+  * @brief JALR - Add 12 bit immediate to source register, save to destination register, jump and link from destination register
+  *
+  * @param[in] _reg  register number (0-31), @param[out] _reg register number (0-31), @param[imm] 12 bit immmediate value
```

Comment:
I fixed this in https://github.com/openucx/ucx/pull/9168/commits/f9083eb7874e0f0dc6f187fac1c54f61e4f97d7a

These are not out variables in the sense that doxygen would consider it. Also there is no "imm" type.